File: GizaWebsite/Components/Dashboard/DashboardController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;

namespace GizaWebsite.Components.Common
{
    public class DashboardController : Controller
    {
        // GET: /Web/

        public ActionResult Index()
        {
            return View("~/Components/Dashboard/dashboard.cshtml");
        }

        public ActionResult Homepage()
        {
            return View("~/Components/Dashboard/Homepage.cshtml");
        }
             
        public ActionResult getLookups()
        {

            try
            {
                 CMSEntities m_context = new CMSEntities();
                 m_context.Configuration.ProxyCreationEnabled = false;

                var sending_entity_types =
                    from row in m_context.ENTITY_TYPE
                    select new 
                      {
                          text = row.NAME_AR
                      };
                var status_types = (from row in m_context.STATUS
                                    select new 
                      {
                          text = row.NAME_AR
                      });
                var document_types = (from row in m_context.DOCUMENT_TYPE
                                                     select new 
                      {
                          text = row.NAME_AR
                      });
                var topic_types = (from row in m_context.TOPIC_TYPE
                                                   select new 
                      {
                          text = row.NAME_AR
                      });
                var adding_entity_types = (from row in m_context.ADDING_ENTITY_TYPE
                                                          select new 
                      {
                          text = row.NAME_AR
                      });
                var c_type = (from row in m_context.C_TYPE
                             select new
                             {
                                 text = row.NAME_AR
                             });
                var importance = (from row in m_context.IMPORTANCEs
                                  select new
                                  {
                                      text = row.NAME_AR
                                  });
                var closets = (from row in m_context.CLOSETs
                                  select new
                                  {
                                      text = row.NAME_AR
                                  });
                var shelves = (from row in m_context.SHELves
                                  select new
                                  {
                                      text = row.NAME_AR
                                  });
                return Json(new object[]{
                               sending_entity_types.ToList() , 
                               status_types.ToList(), 
                               document_types.ToList() ,  
                               topic_types.ToList(),
                               adding_entity_types.ToList(),
                               c_type.ToList(),
                               importance.ToList(),
                               closets.ToList(),
                               shelves.ToList()
                }, JsonRequestBehavior.AllowGet);
            }
            catch (Exception)
            {

                throw;
            }


        }
        public ActionResult saveSendingEntityTypes(List<string> toBeDeleted, List<string> toBeAdded)
        {
            try
            {
                CMSEntities m_context = new CMSEntities();
                m_context.Configuration.ProxyCreationEnabled = false;
                List<string> failedToDelete = new List<string>();
                if (toBeAdded != null)
                {
                    foreach (var item in toBeAdded)
                    {
                        var query = m_context.ENTITY_TYPE.Where(a => a.NAME_AR == item);
                        if (!query.Select(q => q.NAME_AR).Contains(item))
                            m_context.ENTITY_TYPE.Add(new ENTITY_TYPE { NAME_AR = item });
                    }
                    m_context.SaveChanges();
                }
                if (toBeDeleted != null)
                {
                    foreach (var item in toBeDeleted)
                    {
                        try
                        {
                           ENTITY_TYPE s = (ENTITY_TYPE)m_context.ENTITY_TYPE.Where(a => a.NAME_AR == item).First();
                            m_context.ENTITY_TYPE.Remove(s);
                            m_context.SaveChanges();
                        }
                        catch (Exception ex)
                        {
                            failedToDelete.Add(item);
                        }
                    }
                }
                var current  = (from row in m_context.ENTITY_TYPE select row.NAME_AR).ToList();
                return Json(new object[] { failedToDelete, current  }, JsonRequestBehavior.AllowGet);

            }
            catch (Exception)
            {

                throw;
            }


        }
        public ActionResult saveStatus(List<string> toBeDeleted, List<string> toBeAdded)
        {
            try
            {
                CMSEntities m_context = new CMSEntities();
                m_context.Configuration.ProxyCreationEnabled = false;
                List<string> failedToDelete = new List<string>();
                if (toBeAdded != null)
                {
                    foreach (var item in toBeAdded)
                    {
                        var query = m_context.STATUS.Where(a => a.NAME_AR == item);
                        if (!query.Select(q => q.NAME_AR).Contains(item))
                            m_context.STATUS.Add(new STATUS { NAME_AR = item });
                    }
                    m_context.SaveChanges();
                }
                if (toBeDeleted != null)
                {
                    foreach (var item in toBeDeleted)
                    {
                        try
                        {
                            STATUS s = (STATUS)m_context.STATUS.Where(a => a.NAME_AR == item).First();
                            m_context.STATUS.Remove(s);
                            m_context.SaveChanges();
                        }
                        catch (Exception ex)
                        {
                            failedToDelete.Add(item);
                        }
                    }
                }
               
                var currentStatus = (from row in m_context.STATUS select row.NAME_AR).ToList();
                return Json(new object[]{failedToDelete,currentStatus}, JsonRequestBehavior.AllowGet);

            }
            catch (Exception)
            {

                throw;
            }


        }
        public ActionResult saveDocumentTypes(List<string> toBeDeleted, List<string> toBeAdded)
        {
            try
            {
                CMSEntities m_context = new CMSEntities();
                m_context.Configuration.ProxyCreationEnabled = false;
                List<string> failedToDelete = new List<string>();
                if (toBeAdded != null)
                {
                    foreach (var item in toBeAdded)
                    {
                        var query = m_context.DOCUMENT_TYPE.Where(a => a.NAME_AR == item);
                        if (!query.Select(q => q.NAME_AR).Contains(item))
                            m_context.DOCUMENT_TYPE.Add(new DOCUMENT_TYPE { NAME_AR = item });
                    }
                    m_context.SaveChanges();
                }
                if (toBeDeleted != null)
                {
                    foreach (var item in toBeDeleted)
                    {
                        try
                        {
                            DOCUMENT_TYPE s = (DOCUMENT_TYPE)m_context.DOCUMENT_TYPE.Where(a => a.NAME_AR == item).First();
                            m_context.DOCUMENT_TYPE.Remove(s);
                        }
                        catch (Exception ex)
                        {
                            failedToDelete.Add(item);
                        }
                    }
                }
                m_context.SaveChanges();
                var current = (from row in m_context.DOCUMENT_TYPE select row.NAME_AR).ToList();
                return Json(new object[] { failedToDelete, current }, JsonRequestBehavior.AllowGet);

            }
            catch (Exception)
            {

                throw;
            }


        }
        public ActionResult saveTopicTypes(List<string> toBeDeleted, List<string> toBeAdded)
        {
            try
            {
                CMSEntities m_context = new CMSEntities();
                m_context.Configuration.ProxyCreationEnabled = false;
                List<string> failedToDelete = new List<string>();
                if (toBeAdded != null)
                {
                    foreach (var item in toBeAdded)
                    {
                        var query = m_context.TOPIC_TYPE.Where(a => a.NAME_AR == item);
                        if (!query.Select(q => q.NAME_AR).Contains(item))
                            m_context.TOPIC_TYPE.Add(new TOPIC_TYPE { NAME_AR = item });
                    }
                    m_context.SaveChanges();
                }
                if (toBeDeleted != null)
                {
                    foreach (var item in toBeDeleted)
                    {
                        try
                        {
                            TOPIC_TYPE s = (TOPIC_TYPE)m_context.TOPIC_TYPE.Where(a => a.NAME_AR == item).First();
                            m_context.TOPIC_TYPE.Remove(s);
                            m_context.SaveChanges();
                        }
                        catch (Exception ex)
                        {
                            failedToDelete.Add(item);
                        }
                    }
                }
                var current = (from row in m_context.TOPIC_TYPE select row.NAME_AR).ToList();
                return Json(new object[] { failedToDelete, current }, JsonRequestBehavior.AllowGet);

            }
            catch (Exception)
            {

                throw;
            }


        }
        public ActionResult saveAddingEntityTypes(List<string> toBeDeleted, List<string> toBeAdded)
        {
            try
            {
                CMSEntities m_context = new CMSEntities();
                m_context.Configuration.ProxyCreationEnabled = false;
                List<string> failedToDelete = new List<string>();
                if (toBeAdded != null)
                {
                    foreach (var item in toBeAdded)
                    {
                        var query = m_context.ADDING_ENTITY_TYPE.Where(a => a.NAME_AR == item);
                        if (!query.Select(q => q.NAME_AR).Contains(item))
                            m_context.ADDING_ENTITY_TYPE.Add(new ADDING_ENTITY_TYPE { NAME_AR = item });
                    }
                    m_context.SaveChanges();
                }
                if (toBeDeleted != null)
                {
                    foreach (var item in toBeDeleted)
                    {
                        try
                        {
                            ADDING_ENTITY_TYPE s = (ADDING_ENTITY_TYPE)m_context.ADDING_ENTITY_TYPE.Where(a => a.NAME_AR == item).First();
                            m_context.ADDING_ENTITY_TYPE.Remove(s);
                            m_context.SaveChanges();
                        }
                        catch (Exception ex)
                        {
                            failedToDelete.Add(item);
                        }
                    }
                }
                var current = (from row in m_context.ADDING_ENTITY_TYPE select row.NAME_AR).ToList();
                return Json(new object[] { failedToDelete, current }, JsonRequestBehavior.AllowGet);

            }
            catch (Exception)
            {

                throw;
            }


        }
        public ActionResult saveCTypes(List<string> toBeDeleted, List<string> toBeAdded)
        {
            try
            {
                CMSEntities m_context = new CMSEntities();
                m_context.Configuration.ProxyCreationEnabled = false;
                List<string> failedToDelete = new List<string>();
                if (toBeAdded != null)
                {
                    foreach (var item in toBeAdded)
                    {
                        var query = m_context.C_TYPE.Where(a => a.NAME_AR == item);
                        if (!query.Select(q => q.NAME_AR).Contains(item))
                            m_context.C_TYPE.Add(new C_TYPE { NAME_AR = item });
                    }
                    m_context.SaveChanges();
                }
                if (toBeDeleted != null)
                {
                    foreach (var item in toBeDeleted)
                    {
                        try
                        {
                            C_TYPE s = (C_TYPE)m_context.C_TYPE.Where(a => a.NAME_AR == item).First();
                            m_context.C_TYPE.Remove(s);
                            m_context.SaveChanges();
                        }
                        catch (Exception ex)
                        {
                            failedToDelete.Add(item);
                        }
                    }
                }
                var current = (from row in m_context.C_TYPE select row.NAME_AR).ToList();
                return Json(new object[] { failedToDelete, current }, JsonRequestBehavior.AllowGet);

            }
            catch (Exception)
            {

                throw;
            }


        }
        public ActionResult saveImportanceTypes(List<string> toBeDeleted, List<string> toBeAdded)
        {
            try
            {
                CMSEntities m_context = new CMSEntities();
                m_context.Configuration.ProxyCreationEnabled = false;
                List<string> failedToDelete = new List<string>();
                if (toBeAdded != null)
                {
                    foreach (var item in toBeAdded)
                    {
                        var query = m_context.IMPORTANCEs.Where(a => a.NAME_AR == item);
                        if (!query.Select(q => q.NAME_AR).Contains(item))
                            m_context.IMPORTANCEs.Add(new IMPORTANCE { NAME_AR = item });
                    }
                    m_context.SaveChanges();
                }
                if (toBeDeleted != null)
                {
                    foreach (var item in toBeDeleted)
                    {
                        try
                        {
                            IMPORTANCE s = (IMPORTANCE)m_context.IMPORTANCEs.Where(a => a.NAME_AR == item).First();
                            m_context.IMPORTANCEs.Remove(s);
                            m_context.SaveChanges();
                        }
                        catch (Exception ex)
                        {
                            failedToDelete.Add(item);
                        }
                    }
                }
                var current = (from row in m_context.IMPORTANCEs select row.NAME_AR).ToList();
                return Json(new object[] { failedToDelete, current }, JsonRequestBehavior.AllowGet);

            }
            catch (Exception)
            {

                throw;
            }


        }

        public ActionResult saveClosets(List<string> toBeDeleted, List<string> toBeAdded)
        {
            try
            {
                CMSEntities m_context = new CMSEntities();
                m_context.Configuration.ProxyCreationEnabled = false;
                List<string> failedToDelete = new List<string>();
                if (toBeAdded != null)
                {
                    foreach (var item in toBeAdded)
                    {
                        var query = m_context.CLOSETs.Where(a => a.NAME_AR == item);
                        if (!query.Select(q => q.NAME_AR).Contains(item))
                            m_context.CLOSETs.Add(new CLOSET { NAME_AR = item });
                    }
                    m_context.SaveChanges();
                }
                if (toBeDeleted != null)
                {
                    foreach (var item in toBeDeleted)
                    {
                        try
                        {
                            CLOSET s = (CLOSET)m_context.CLOSETs.Where(a => a.NAME_AR == item).First();
                            m_context.CLOSETs.Remove(s);
                            m_context.SaveChanges();
                        }
                        catch (Exception ex)
                        {
                            failedToDelete.Add(item);
                        }
                    }
                }
                var current = (from row in m_context.CLOSETs select row.NAME_AR).ToList();
                return Json(new object[] { failedToDelete, current }, JsonRequestBehavior.AllowGet);

            }
            catch (Exception)
            {

                throw;
            }


        }

        public ActionResult saveShelves(List<string> toBeDeleted, List<string> toBeAdded)
        {
            try
            {
                CMSEntities m_context = new CMSEntities();
                m_context.Configuration.ProxyCreationEnabled = false;
                List<string> failedToDelete = new List<string>();
                if (toBeAdded != null)
                {
                    foreach (var item in toBeAdded)
                    {
                        var query = m_context.SHELves.Where(a => a.NAME_AR == item);
                        if (!query.Select(q => q.NAME_AR).Contains(item))
                            m_context.SHELves.Add(new SHELF { NAME_AR = item });
                    }
                    m_context.SaveChanges();
                }
                if (toBeDeleted != null)
                {
                    foreach (var item in toBeDeleted)
                    {
                        try
                        {
                            SHELF s = (SHELF)m_context.SHELves.Where(a => a.NAME_AR == item).First();
                            m_context.SHELves.Remove(s);
                            m_context.SaveChanges();
                        }
                        catch (Exception ex)
                        {
                            failedToDelete.Add(item);
                        }
                    }
                }
                var current = (from row in m_context.SHELves select row.NAME_AR).ToList();
                return Json(new object[] { failedToDelete, current }, JsonRequestBehavior.AllowGet);

            }
            catch (Exception)
            {

                throw;
            }


        }
        public ActionResult saveHomePageContent(string htmlContent)
        {
            if (!string.IsNullOrEmpty(htmlContent))
            {
                try
                {
                      CMSEntities m_context = new CMSEntities();
                      m_context.Configuration.ProxyCreationEnabled = false;
                      var query = from rows in m_context.HOMEPAGE_CONTENT select rows;
                      foreach (var item in query)
                          m_context.HOMEPAGE_CONTENT.Remove(item);
                      m_context.HOMEPAGE_CONTENT.Add(new HOMEPAGE_CONTENT { HTML_CONTENT = htmlContent });
                      m_context.SaveChanges();
                      return Json(new object[] { 1 }, JsonRequestBehavior.AllowGet);

                }
                catch (Exception)
                {
                    return Json(new object[] { 0 }, JsonRequestBehavior.AllowGet);
                }
            }
            else
                return Json(new object[] { 0 }, JsonRequestBehavior.AllowGet);

        }
        public ActionResult getHomePageContent()
        {
            HomePage.HomeController c = new HomePage.HomeController();
            return c.getHomePageContent();
        }
    }
}

File: GizaWebsite/Components/Dashboard/DashboardController.js
App.controller('DashboardController', ["Utility", "api", "$scope", "$rootScope", "$window", "$timeout",
    function (Utility, api, $scope, $rootScope, $window, $timeout) {

        if (sessionStorage.isAdmin == "false")
            $window.location.href = "/home/noaccess";

        $scope.flag = false;
        $scope.errorflag = false;
        $scope.sending_entity_types = [];
        $scope.status_types = [];
        $scope.document_types = [];
        $scope.topic_types = [];
        $scope.adding_entity_types = [];
        $scope.c_types = [];
        $scope.importance_types = [];
        $scope.closets = [];
        $scope.shelves = [];

        var sending_entity_types = [];
        var status_types = [];
        var document_types = [];
        var topic_types = [];
        var adding_entity_types = [];
        var c_types = [];
        var importance_types = [];
        var closets = [];
        var shelves = [];

        $scope.getLookups = function () {
            $("#loading").css("display", "block");
            api.Get("/Dashboard/getLookups").success(function (Response) {
                if (Response[0]) {
                    sending_entity_types = angular.copy(Response[0]);
                    $scope.sending_entity_types = Response[0];
                }
                if (Response[1]) {
                    status_types = angular.copy(Response[1]);
                    $scope.status_types = Response[1];
                }
                if (Response[2]) {
                    document_types = angular.copy(Response[2]);
                    $scope.document_types = Response[2];
                }
                if (Response[3]) {
                    topic_types = angular.copy(Response[3]);
                    $scope.topic_types = Response[3];
                }
                if (Response[4]) {
                    adding_entity_types = angular.copy(Response[4]);
                    $scope.adding_entity_types = Response[4];
                }
                if (Response[5]) {
                    c_types = angular.copy(Response[5]);
                    $scope.c_types = Response[5];
                }
                if (Response[6]) {
                    importance_types = angular.copy(Response[6]);;
                    $scope.importance_types = Response[6];
                }
                if (Response[7]) {
                    closets = angular.copy(Response[7]);;
                    $scope.closets = Response[7];
                }
                if (Response[8]) {
                    shelves = angular.copy(Response[8]);;
                    $scope.shelves = Response[8];
                }
                $("#loading").css("display", "none");
            }).error(function (error) { console.log(error); })
        }
        $scope.getAdditions = function (before, after) {
            var toBeAdded = [];
            for (var i = 0; i < after.length; i++) {
                var found = false;
                for (var j = 0; j < before.length; j++) {
                    if (after[i].text == before[j].text) {
                        found = true;
                        break;
                    }
                }
                if (!found)
                    toBeAdded.push(after[i].text);
            }
            return toBeAdded;
        }
        $scope.getDeletions = function (before, after) {
            var toBeDeleted = [];
            for (var i = 0; i < before.length; i++) {
                var found = false;
                for (var j = 0; j < after.length; j++) {
                    if (after[j].text == before[i].text) {
                        found = true;
                        break;
                    }
                }
                if (!found)
                    toBeDeleted.push(before[i].text);
            }
            return toBeDeleted;
        }

        $scope.saveSendingEntityTypes = function () {
            var toBeAdded = $scope.getAdditions(sending_entity_types, $scope.sending_entity_types);
            var toBeDeleted = $scope.getDeletions(sending_entity_types, $scope.sending_entity_types);
            api.Post("/Dashboard/saveSendingEntityTypes", null, { toBeAdded: toBeAdded, toBeDeleted: toBeDeleted }).success(function (Response) {
                if (Response[0]) {
                    if (Response[0].length > 0) {
                        var str = '';
                        for (var i = 0; i < Response[0].length; i++) {
                            str += Response[i] + ',';
                        }
                        str = str.substring(0, str.length - 1);
                        $('#errormsg').html(' ":لم نتمكن من حذف الحقول' + str + ' "لارتباطها بعناصر');
                        $scope.flag = false;
                        $scope.errorflag = false;
                        $timeout(function () {
                            $scope.errorflag = true;
                        }, 1000);
                    }
                    else {
                        if (Response[1]) {
                            $scope.flag = false;
                            $scope.errorflag = false;
                            $timeout(function () {
                                $scope.flag = true;
                            }, 1000);
                            $scope.sending_entity_types = angular.copy(Response[1]);
                        }
                    }
                }
                $("#scroll_top").focus();
            }).error(function (error) { console.log(error); })
        }
        $scope.saveStatus = function () {
            var toBeAdded = $scope.getAdditions(status_types, $scope.status_types);
            var toBeDeleted = $scope.getDeletions(status_types, $scope.status_types);
            api.Post("/Dashboard/saveStatus", null, { toBeAdded: toBeAdded, toBeDeleted: toBeDeleted }).success(function (Response) {

                if (Response[0]) {
                    if (Response[0].length > 0) {
                        var str = '';
                        for (var i = 0; i < Response[0].length; i++) {
                            str += Response[i] + ',';
                        }
                        str = str.substring(0, str.length - 1);
                        $('#errormsg').html(' :لم نتمكن من حذف الحقول' + str + ' لارتباطها بعناصر');
                        $scope.flag = false;
                        $scope.errorflag = false;
                        $timeout(function () {
                            $scope.errorflag = true;
                        }, 1000);
                    }
                    else {
                        if (Response[1]) {
                            $scope.flag = false;
                            $scope.errorflag = false;
                            $timeout(function () {
                                $scope.flag = true;
                            }, 1000);
                            $scope.status_types = angular.copy(Response[1]);
                        }
                    }
                }
                $("#scroll_top").focus();
            }).error(function (error) { console.log(error); })
        }
        $scope.saveDocumentTypes = function () {
            var toBeAdded = $scope.getAdditions(document_types, $scope.document_types);
            var toBeDeleted = $scope.getDeletions(document_types, $scope.document_types);
            api.Post("/Dashboard/saveDocumentTypes", null, { toBeAdded: toBeAdded, toBeDeleted: toBeDeleted }).success(function (Response) {
                if (Response[0]) {
                    if (Response[0].length > 0) {
                        var str = '';
                        for (var i = 0; i < Response[0].length; i++) {
                            str += Response[i] + ',';
                        }
                        str = str.substring(0, str.length - 1);
                        $('#errormsg').html(' :لم نتمكن من حذف الحقول' + str + ' لارتباطها بعناصر');
                        $scope.flag = false;
                        $scope.errorflag = false;
                        $timeout(function () {
                            $scope.errorflag = true;
                        }, 1000);
                    }
                    else {
                        if (Response[1]) {
                            $scope.flag = false;
                            $scope.errorflag = false;
                            $timeout(function () {
                                $scope.flag = true;
                            }, 1000);
                            $scope.document_types = angular.copy(Response[1]);
                        }
                    }
                }
                $("#scroll_top").focus();
            }).error(function (error) { console.log(error); })
        }
        $scope.saveTopicTypes = function () {
            var toBeAdded = $scope.getAdditions(topic_types, $scope.topic_types);
            var toBeDeleted = $scope.getDeletions(topic_types, $scope.topic_types);
            api.Post("/Dashboard/saveTopicTypes", null, { toBeAdded: toBeAdded, toBeDeleted: toBeDeleted }).success(function (Response) {
                if (Response[0]) {
                    if (Response[0].length > 0) {
                        var str = '';
                        for (var i = 0; i < Response[0].length; i++) {
                            str += Response[i] + ',';
                        }
                        str = str.substring(0, str.length - 1);
                        $('#errormsg').html(' :لم نتمكن من حذف الحقول' + str + ' لارتباطها بعناصر');
                        $scope.flag = false;
                        $scope.errorflag = false;
                        $timeout(function () {
                            $scope.errorflag = true;
                        }, 1000);
                    }
                    else {
                        if (Response[1]) {
                            $scope.flag = false;
                            $scope.errorflag = false;
                            $timeout(function () {
                                $scope.flag = true;
                            }, 1000);
                            $scope.topic_types = angular.copy(Response[1]);
                        }
                    }
                }
                $("#scroll_top").focus();
            }).error(function (error) { console.log(error); })
        }
        $scope.saveAddingEntityTypes = function () {
            var toBeAdded = $scope.getAdditions(adding_entity_types, $scope.adding_entity_types);
            var toBeDeleted = $scope.getDeletions(adding_entity_types, $scope.adding_entity_types);
            api.Post("/Dashboard/saveAddingEntityTypes", null, { toBeAdded: toBeAdded, toBeDeleted: toBeDeleted }).success(function (Response) {
                if (Response[0]) {
                    if (Response[0].length > 0) {
                        var str = '';
                        for (var i = 0; i < Response[0].length; i++) {
                            str += Response[i] + ',';
                        }
                        str = str.substring(0, str.length - 1);
                        $('#errormsg').html(' :لم نتمكن من حذف الحقول' + str + ' لارتباطها بعناصر');
                        $scope.flag = false;
                        $scope.errorflag = false;
                        $timeout(function () {
                            $scope.errorflag = true;
                        }, 1000);
                    }
                    else {
                        if (Response[1]) {
                            $scope.flag = false;
                            $scope.errorflag = false;
                            $timeout(function () {
                                $scope.flag = true;
                            }, 1000);
                            $scope.adding_entity_types = angular.copy(Response[1]);
                        }
                    }
                }
                $("#scroll_top").focus();
            }).error(function (error) { console.log(error); })
        }
        $scope.saveCTypes = function () {
            var toBeAdded = $scope.getAdditions(c_types, $scope.c_types);
            var toBeDeleted = $scope.getDeletions(c_types, $scope.c_types);
            api.Post("/Dashboard/saveCTypes", null, { toBeAdded: toBeAdded, toBeDeleted: toBeDeleted }).success(function (Response) {
                if (Response[0]) {
                    if (Response[0].length > 0) {
                        var str = '';
                        for (var i = 0; i < Response[0].length; i++) {
                            str += Response[i] + ',';
                        }
                        str = str.substring(0, str.length - 1);
                        $('#errormsg').html(' :لم نتمكن من حذف الحقول' + str + ' لارتباطها بعناصر');
                        $scope.flag = false;
                        $scope.errorflag = false;
                        $timeout(function () {
                            $scope.errorflag = true;
                        }, 1000);
                    }
                    else {
                        if (Response[1]) {
                            $scope.flag = false;
                            $scope.errorflag = false;
                            $timeout(function () {
                                $scope.flag = true;
                            }, 1000);
                            $scope.c_types = angular.copy(Response[1]);
                        }
                    }
                }
                $("#scroll_top").focus();
            }).error(function (error) { console.log(error); })
        }
        $scope.saveImportanceTypes = function () {
            var toBeAdded = $scope.getAdditions(importance_types, $scope.importance_types);
            var toBeDeleted = $scope.getDeletions(importance_types, $scope.importance_types);
            api.Post("/Dashboard/saveImportanceTypes", null, { toBeAdded: toBeAdded, toBeDeleted: toBeDeleted }).success(function (Response) {
                if (Response[0]) {
                    if (Response[0].length > 0) {
                        var str = '';
                        for (var i = 0; i < Response[0].length; i++) {
                            str += Response[i] + ',';
                        }
                        str = str.substring(0, str.length - 1);
                        $('#errormsg').html(' :لم نتمكن من حذف الحقول' + str + ' لارتباطها بعناصر');
                        $scope.flag = false;
                        $scope.errorflag = false;
                        $timeout(function () {
                            $scope.errorflag = true;
                        }, 1000);
                    }
                    else {
                        if (Response[1]) {
                            $scope.flag = false;
                            $scope.errorflag = false;
                            $timeout(function () {
                                $scope.flag = true;
                            }, 1000);
                            $scope.importance_types = angular.copy(Response[1]);
                        }
                    }
                }
                $("#scroll_top").focus();
            }).error(function (error) { console.log(error); })
        }
        $scope.saveClosets = function () {
            var toBeAdded = $scope.getAdditions(closets, $scope.closets);
            var toBeDeleted = $scope.getDeletions(closets, $scope.closets);
            api.Post("/Dashboard/saveClosets", null, { toBeAdded: toBeAdded, toBeDeleted: toBeDeleted }).success(function (Response) {
                if (Response[0]) {
                    if (Response[0].length > 0) {
                        var str = '';
                        for (var i = 0; i < Response[0].length; i++) {
                            str += Response[i] + ',';
                        }
                        str = str.substring(0, str.length - 1);
                        $('#errormsg').html(' :لم نتمكن من حذف الحقول' + str + ' لارتباطها بعناصر');
                        $scope.flag = false;
                        $scope.errorflag = false;
                        $timeout(function () {
                            $scope.errorflag = true;
                        }, 1000);
                    }
                    else {
                        if (Response[1]) {
                            $scope.flag = false;
                            $scope.errorflag = false;
                            $timeout(function () {
                                $scope.flag = true;
                            }, 1000);
                            $scope.closets = angular.copy(Response[1]);
                        }
                    }
                }
                $("#scroll_top").focus();
            }).error(function (error) { console.log(error); })
        }
        $scope.saveShelves = function () {
            var toBeAdded = $scope.getAdditions(shelves, $scope.shelves);
            var toBeDeleted = $scope.getDeletions(shelves, $scope.shelves);
            api.Post("/Dashboard/saveShelves", null, { toBeAdded: toBeAdded, toBeDeleted: toBeDeleted }).success(function (Response) {
                if (Response[0]) {
                    if (Response[0].length > 0) {
                        var str = '';
                        for (var i = 0; i < Response[0].length; i++) {
                            str += Response[i] + ',';
                        }
                        str = str.substring(0, str.length - 1);
                        $('#errormsg').html(' :لم نتمكن من حذف الحقول' + str + ' لارتباطها بعناصر');
                        $scope.flag = false;
                        $scope.errorflag = false;
                        $timeout(function () {
                            $scope.errorflag = true;
                        }, 1000);
                    }
                    else {
                        if (Response[1]) {
                            $scope.flag = false;
                            $scope.errorflag = false;
                            $timeout(function () {
                                $scope.flag = true;
                            }, 1000);
                            $scope.shelves = angular.copy(Response[1]);
                        }
                    }
                }
                $("#scroll_top").focus();
            }).error(function (error) { console.log(error); })
        }

        $scope.getLookups();
    }]);
